### PR TITLE
fix: fix content type when saving files

### PR DESCRIPTION
**Pull request by scottdover**
**Summary**
Previously, saving files would always convert the content type to text/plain. This changes things such that we maintain the content type on save.

Additionally, this makes a few small refactoring changes:
 - This moves a couple of helper functions from ContentModel to utils
 - It removes some unnecessary writes to file metadata

**Testing**
 - [x] Test creating file
 - [x] Test saving file
 - [x] Test renaming file

### CODE REVIEW

File: client/src/components/ContentNavigator/ContentModel.ts [hunk 1]
```diff
@@ -362,20 +336,21 @@ export class ContentModel {
...
   } {
-    if (resourceId in this.fileTokenMaps) {
-      return this.fileTokenMaps[resourceId];
+    if (resourceId in this.fileMetadataMap) {
+      return this.fileMetadataMap[resourceId];
     }
     const now = new Date();
     const timestamp = now.toUTCString();
-    return { etag: "", lastModified: timestamp };
+    return { etag: "", lastModified: timestamp, contentType: "text/plain" };
```
Comment (scnwwu) [inline client/src/components/ContentNavigator/ContentModel.ts:346]: nit: DEFAULT_FILE_CONTENT_TYPE

File: client/src/components/ContentNavigator/ContentModel.ts [hunk 2]
```diff
@@ -244,16 +227,13 @@ export class ContentModel {
   ): Promise<ContentItem | undefined> {
     const contentType = await this.getFileContentType(fileName);
```
Comment (scnwwu) [inline client/src/components/ContentNavigator/ContentModel.ts:228]: nit: it looks a little confusing with `fetchFileContentType`. It's hard to tell the difference from the name "fetch" vs. "get". The implementation look unrelated to each other.

Comment (scottdover) [reply client/src/components/ContentNavigator/ContentModel.ts]: I updated the method to better match what is being returned, and updating both "fetch" methods to use a prefix of "get" since that seems to be more consistent with what we use elsewhere https://github.com/sassoftware/vscode-sas-extension/pull/980/commits/1ee77cc65681082fc3b202e55e265821d23a11db

File: client/src/components/ContentNavigator/ContentModel.ts (hunk 1 above)
Comment (scottdover) [reply client/src/components/ContentNavigator/ContentModel.ts]: Done in https://github.com/sassoftware/vscode-sas-extension/pull/980/commits/1ee77cc65681082fc3b202e55e265821d23a11db

**Review by scnwwu — APPROVED** [review]

**Review by ShermanWang0324 — APPROVED** [review]
